apps/student-app/src/app/student-app.tsx

```diff
@@ -0,0 +1,379 @@
+  }
+
+  // to populate the LI elements for the word lists ()
+  // Idea is to set the word list here, then confirm later in the form
+  function populate_word_list_items() {
+    return data?.word_lists.map((word_list: WordList) => {
+      return (
+        <li
+          id={word_list.id} key={word_list.id}
+        >
+          <button id={`word-list-${word_list.id}`} onClick={() => handleSetWordList(word_list.id)}>
+            <b>{word_list.name}:</b> Words: {word_list.words.length}
+            </button>
+        </li>
+      );
+    });
+
+  }
+
+  // Creates a list of words that were in the game.
+  // Populates a UL element on game-words page.
+  function get_words_from_game() {
+    return words_in_game.map((word: Word) => {
+      return (
+        <li id={`word-${word.word}`}  key={`word-${word.word}`}>
+          <b className={words_correct.includes(word) ? 'correct-word' : 'incorrect-word'}>{word.word}:</b> {word.hint}
+        </li>
+      );
+    });
+  }
+
+  // Button to go to Crouse page
+  function handleToCourse(e: any): void {
+    try {
+      setError('null');
+      setScreen(StudentState.course_state);
+    } catch (err) {
+      setError('error-word-list');
+    }
+  }
+
+  // Back button to return to Splash page
+  function handleBackToSplash(e: any): void {
+    try {
+      setError('null');
+      setScreen(StudentState.splash_state);
+    } catch (err) {
+      setError('error-course');
+    }
+  }
+
+  // Back button to return to Course Code page
+  function handleBackToCourse(e: any): void {
+    try {
+      setError('null');
+      setScreen(StudentState.course_state);
+    } catch (err) {
+      setError('error-word-list');
+    }
+  }
+
+  // Back button to return to Word List page
+  function handleBackToWordLists(e: any): void {
+    try {
+      setError('null');
+      setScreen(StudentState.word_list_state);
+    } catch (err) {
+      setError('error-word-list');
+    }
+  }
+
+  // Back button to return to Score page
+  function handleBackToScore(e: any): void {
+    try {
+      setError('null');
+      setScreen(StudentState.score_state);
+    } catch (err) {}
+  }
+
+  // Replay button that resets the game and starts it again.
+  function handleResetGame(e: any): void {
+    try {
+      setError('null');
+      start_new_game();
+      setScreen(StudentState.game_state);
+    } catch (err) {}
+  }
+
+  // Button to visit Word Review page
+  function handleToGameWords(e: any): void {
+    try {
+      setError('null');
+      setScreen(StudentState.review_words_state);
+    } catch (err) {}
+  }
+
+  // Handles course code submission
+  async function handleSubmitCourse(e: React.FormEvent<HTMLFormElement>): Promise<void> {
+    e.preventDefault();
+    try {
+      // GET on course GUID
+      // STUB getting the data
+      setData(await submitCourse(course_code));
+      populate_word_list_items(); //Create the word lists List Items elements
```

Comment:
this should not be called here, it should be called on inside of rendering code

Reply:
`populate_word_list_items();` does nothing in this place. so yeah in the html within the return (it is already there), so this can be removed, so just pointing out it is just not supposed to be in this spot.

Reply:
done